messaging/src/main/java/org/axonframework/eventhandling/scheduling/dbscheduler/DbSchedulerEventScheduler.java

```diff
@@ -0,0 +1,281 @@
+import org.axonframework.serialization.SimpleSerializedObject;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicReference;
+import javax.annotation.Nonnull;
+
+import static java.util.Objects.isNull;
+import static org.axonframework.common.BuilderUtils.assertNonNull;
+import static org.axonframework.eventhandling.GenericEventMessage.clock;
+import static org.axonframework.eventhandling.scheduling.dbscheduler.DbSchedulerScheduleToken.TASK_NAME;
+
+/**
+ * EventScheduler implementation that delegates scheduling and triggering to a db-scheduler Scheduler.
+ *
+ * @author Gerard Klijs
+ * @since 4.8.0
+ */
+@SuppressWarnings("Duplicates")
+public class DbSchedulerEventScheduler implements EventScheduler, Lifecycle {
+
+    private static final AtomicReference<DbSchedulerEventScheduler> eventSchedulerReference = new AtomicReference<>();
+    private final Scheduler scheduler;
+    private final Serializer serializer;
+    private final TransactionManager transactionManager;
+    private final EventBus eventBus;
+
+    /**
+     * Instantiate a {@link DbSchedulerEventScheduler} based on the fields contained in the
+     * {@link DbSchedulerEventScheduler.Builder}.
+     * <p>
+     * Will assert that the {@link Scheduler}, {@link Serializer} and {@link EventBus} are not {@code null}, and will
+     * throw an {@link AxonConfigurationException} if any of them is {@code null}.
+     *
+     * @param builder the {@link Builder} used to instantiate a {@link DbSchedulerEventScheduler} instance
+     */
+    protected DbSchedulerEventScheduler(Builder builder) {
+        builder.validate();
+        scheduler = builder.scheduler;
+        serializer = builder.serializer;
+        transactionManager = builder.transactionManager;
+        eventBus = builder.eventBus;
+        eventSchedulerReference.set(this);
+    }
+
+    /**
+     * Instantiate a Builder to be able to create a {@link DbSchedulerEventScheduler}.
+     * <p>
+     * The {@link TransactionManager} is defaulted to a {@link NoTransactionManager}.
+     * <p>
+     * The {@link Scheduler}, {@link Serializer} and {@link EventBus} are <b>hard requirements</b> and as such should be
+     * provided.
+     *
+     * @return a Builder to be able to create a {@link DbSchedulerEventScheduler}
+     */
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    @Override
+    public ScheduleToken schedule(Instant triggerDateTime, Object event) {
+        DbSchedulerScheduleToken taskInstanceId = new DbSchedulerScheduleToken(UUID.randomUUID().toString());
+        try {
+            DbSchedulerEventData data;
+            if (event instanceof EventMessage) {
+                data = detailsFromEvent((EventMessage<?>) event);
+            } else {
+                data = detailsFromObject(event);
+            }
+            TaskInstance<?> taskInstance = task().instance(taskInstanceId.getId(), data);
```

Comment:
If you want to can create the task-instance without a reference to the actual task. For example similar to this: 

https://github.com/kagkarlsson/db-scheduler/blob/master/examples/spring-boot-example/src/main/java/com/github/kagkarlsson/examples/boot/config/TransactionallyStagedJobConfiguration.java#L48

Reply:
Thanks, that looks cleaner. I changed the code accordingly.